docker/Dockerfile.isce3.dev

```diff
@@ -0,0 +1,84 @@
+# A Dockerfile to install ISCE3 from source code,
+# so that the up-to-date features can be tested.
+#
+# Instruction
+#
+
+FROM oraclelinux:8
+
+LABEL author="OPERA ADT" \
+    description="s1 cslc 0.2 Gamma release" \
+    version="0.2-gamma"
+
+RUN yum -y update &&\
+    yum -y install curl git make unzip &&\
+    adduser compass_user &&\
+    mkdir -p /.config && chmod -R 777 /.config &&\
+    mkdir -p /.cache && chmod -R 777 /.cache
+
+USER compass_user
+
+RUN mkdir -p /home/compass_user/OPERA &&\
+    chmod -R 755 /home/compass_user
+
+ENV CONDA_PREFIX=/home/compass_user/miniconda3
+
+# install Miniconda
+WORKDIR /home/compass_user
+RUN curl -sSL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o miniconda.sh &&\
+    bash miniconda.sh -b -p ${CONDA_PREFIX} &&\
+    rm $HOME/miniconda.sh
+
+ENV PATH=${CONDA_PREFIX}/bin:${PATH}
+RUN ${CONDA_PREFIX}/bin/conda init bash
+
+# copy COMPASS and set compass_user as owner
+COPY --chown=compass_user:compass_user . /home/compass_user/OPERA/COMPASS
+
+# create CONDA environment
+RUN conda create --name "COMPASS" --file /home/compass_user/OPERA/COMPASS/docker/specfile.isce3.dev.txt
+
+SHELL ["conda", "run", "-n", "COMPASS", "/bin/bash", "-c"]
+
+# Install ISCE3
+RUN mkdir -p /home/compass_user/OPERA/ISCE3/install &&\
+    mkdir -p /home/compass_user/OPERA/ISCE3/build &&\
+    cd /home/compass_user/OPERA/ISCE3/build &&\
+    cmake -DCMAKE_INSTALL_PREFIX=/home/compass_user/OPERA/ISCE3/install -DWITH_CUDA=OFF /home/compass_user/OPERA/COMPASS/isce-develop &&\
+    make -j32 VERBOSE=ON &&\
```

Comment:
How do we know there will always be 32 threads? `-j32` is not ideal for anyone not using a 32-thread machine.

I think the `nproc` command could be used here to address this. Before this install ISCE3 command add:
```
ENV N_THREADS=nproc
```
and the `make` command above becomes:
```
make -j${N_THREADS} VERBOSE=ON &&\
```
Sorry untested.

Reply:
Thanks for the suggestion @LiangJYu. I think that is more general than what I've hard-coded here.
One concern is that I do not want to use all CPU threads int he system. Is there a way to pass half or quarter of `nproc`?

Reply:
Closing this comment as I've take out the on-the-fly ISCE3 builder